### DISCUSSION TITLE
Recommended result codes

### DISCUSSION THREAD

**Discussion by DenisAnisimov** [dokan-dev/dokany#1160] [Q&A] [answered]
Are there any recommended result codes that I should return when calling my ReadFile callback in a situation where:

1) Less data is available for reading than requested.
2) Offset is greater than file size.

Should I somehow handle the PagingIo and SynchronousIo values in my ReadFile callback?

**Comment by Liryna** [comment, marked as answer]
Hi @DenisAnisimov ,
Both cases are handled in memfs sample. You can see the values returned there.
